Logstash/logstash-input-azureeventhub/lib/logstash/inputs/azureeventhub.rb

```diff
@@ -40,43 +40,76 @@ def register
   end # def register
 
   def get_pay_load(message)
-    return nil if not message
+    return nil, nil if not message
+    annotationMap = nil
+    body = nil
     message.getPayload().each do |section|
-      if section.java_kind_of? org::apache::qpid::amqp_1_0::type::messaging::Data or section.java_kind_of? org::apache::qpid::amqp_1_0::type::messaging::AmqpValue
-        return section.getValue().to_s.gsub("\\x5c", "\\")
+      if section.java_kind_of? org::apache::qpid::amqp_1_0::type::messaging::MessageAnnotations
+        annotationMap = section.getValue()
+      else
+        if section.java_kind_of? org::apache::qpid::amqp_1_0::type::messaging::Data or section.java_kind_of? org::apache::qpid::amqp_1_0::type::messaging::AmqpValue
+          body = section.getValue().to_s.gsub("\\x5c", "\\")
+        end
       end
     end
-    return nil
+    return body, annotationMap
   end
   
-  def process(output_queue, receiver, partition)
+  def process(output_queue, receiver, partition, last_event_offset)
     while !stop?
       begin
         msg = receiver.receive(10)
         if msg
-          codec.decode(get_pay_load(msg)) do |event|
+          body, annotationMap = get_pay_load(msg)
+          last_event_offset = annotationMap.get(org::apache::qpid::amqp_1_0::type::Symbol.valueOf("x-opt-offset")) unless annotationMap.nil?
+          @logger.debug("[#{partition.to_s.rjust(2,"0")}] Event: #{body[0..50] unless body.nil?}... " +
+            "Offset: #{annotationMap.get(org::apache::qpid::amqp_1_0::type::Symbol.valueOf("x-opt-offset")) unless annotationMap.nil? } " +
+            "Time: #{annotationMap.get(org::apache::qpid::amqp_1_0::type::Symbol.valueOf("x-opt-enqueued-time")).to_s unless annotationMap.nil? } " +
```

Comment:
nit: Consider use << instead of + for string appending.

Reply:
fixed